internal/panda_controller/docker_grpc_agent.go

```diff
@@ -129,10 +129,25 @@ func (pa *dockerGrpcPandaAgent) StopRecording(ctx context.Context) (*PandaAgentR
 	// Strip off the shared folder prefix so that paths are correct on this system
 	recordingName := strings.Replace(recording.RecordingName, "./shared/", "", 1)
 
-	return &PandaAgentRecording{
+	new_recording := PandaAgentRecording{
 		RecordingName: recordingName,
-		Location: *pa.sharedDir,
-	}, nil
+		Location:      *pa.sharedDir,
+	}
+
+	// Copy given image into shared directory
+	//TODO: Replace destination with filesystem target
+	nBytes, err := copyFileHelper(new_recording.GetSnapshotFileName(), "/tmp/panda-studio/", fmt.Sprintf("%s-rr-snp", new_recording.RecordingName))
+	if (err != nil && err != io.EOF) || nBytes == 0 {
+		print("Error in copying snapshot")
```

Comment:
Please keep print statements to a minimum; If you like, you can use `errors.Wrap` - documentation here https://pkg.go.dev/github.com/pkg/errors?utm_source=godoc to add additional context